### PR TITLE
tiny js worls is ready for review

### PR DESCRIPTION
**Pull request by m-ruslan**
Tiny JS World

[Demo](https://m-ruslan.github.io/a-tiny-JS-world/) |
[Code base](https://github.com/m-ruslan/a-tiny-JS-world)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@m-ruslan please keep only `index.js` in this PR. Other files aren't subject to code review

**Comment by m-ruslan** [comment]
@OleksiyRudenko thank you
I've deleted all other files. Could you please review?

File: submissions/m-ruslan/a-tiny-JS-world/index.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+   Message can contain HTML markup. You may also tweak index.html and/or styles.css.
+   However, please, REFRAIN from improving visuals at least until your code is reviewed
+   so code reviewers might focus on a single file that is index.js.
+   */
+
+let makeMassege = (obj) =>
+  `${obj.type}; ${obj.legs}; ${obj.hands}; ${obj.name}; ${
+    obj.saying
+  }; ${obj.friendTo.join(",")}`;
```
Comment (OleksiyRudenko) [inline submissions/m-ruslan/a-tiny-JS-world/index.js:53]: Let's employ Array methods `map` and `join` to improve readability and scalability here.

Hint: `[a, b, c.join()].join()` (well formatted though)

File: submissions/m-ruslan/a-tiny-JS-world/index.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+let world = [dog, cat, woman, man];
+world.forEach((e) => print(makeMassege(e)));
```
Comment (OleksiyRudenko) [inline submissions/m-ruslan/a-tiny-JS-world/index.js:56]: `e` is not as representative variable as e.g. `cat`, `woman`, `world` etc
Can you devise a name that would represent iterated objects better?

P.S. There is also a typo in `makeMassege`. You must have meant `makeMessage`

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@m-ruslan well done.
Let's polish it and we're done.

File: submissions/m-ruslan/a-tiny-JS-world/index.js (hunk 1 above)
Comment (m-ruslan) [reply submissions/m-ruslan/a-tiny-JS-world/index.js]: Thank you.
I've made changes. Please review)

File: submissions/m-ruslan/a-tiny-JS-world/index.js (hunk 2 above)
Comment (m-ruslan) [reply submissions/m-ruslan/a-tiny-JS-world/index.js]: Thank you.
I've changed this

File: submissions/m-ruslan/a-tiny-JS-world/index.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+/* Use print(message) for output.
+   Default tag for message is <pre>. Use print(message,'div') to change containing element tag.
+
+   Message can contain HTML markup. You may also tweak index.html and/or styles.css.
+   However, please, REFRAIN from improving visuals at least until your code is reviewed
+   so code reviewers might focus on a single file that is index.js.
+   */
+
+let makeMessage = (obj) =>
+  Object.keys(obj)
```
Comment (OleksiyRudenko) [inline submissions/m-ruslan/a-tiny-JS-world/index.js:51]: You had explicit list of properties. With `Object.keys` it became implicit, which is not really good for some reasons:
- we want to know what exactly properties gonna be printed
- we may want to control what exactly props to use and what to skip
- order of keys is not guaranteed

Please fix.

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@m-ruslan you managed to employ `Array` methods, but using `Object.keys` made properties use implicit. Let's fix this.

File: submissions/m-ruslan/a-tiny-JS-world/index.js (hunk 3 above)
Comment (m-ruslan) [reply submissions/m-ruslan/a-tiny-JS-world/index.js]: Oleksiy, thank you
I've already sent to review task ["post-OOP"](https://github.com/kottans/frontend-2021-homeworks/pull/220) and changed there `Object.keys` to `for...in` loop, but I'm not sure that this is the right answer here (also added `for...in` loop to this PR).

Could you please check one more time?)

File: submissions/m-ruslan/a-tiny-JS-world/index.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+   so code reviewers might focus on a single file that is index.js.
+   */
+
+let makeMessage = (obj) => {
+  let propsArr = [];
+  for (let propName in obj) {
+    propName !== "constructor"
+      ? propsArr.push(propName)
+      : console.log("constructor found");
+  }
```
Comment (OleksiyRudenko) [inline submissions/m-ruslan/a-tiny-JS-world/index.js:56]: Statement "Order of object keys is not guaranteed" is still aplicable.
`props = [obj.a, obj.b, ...]` would be explicit.
`['a', 'b'].map(propName => obj[propName])` is explicit and less code repetition.

File: submissions/m-ruslan/a-tiny-JS-world/index.js [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+  return propsArr.map((propName) => obj[propName]).join("; ");
```
Comment (OleksiyRudenko) [inline submissions/m-ruslan/a-tiny-JS-world/index.js:57]: Your callback may be as short as 
`.map(obj => [... propnames ...].map(...).join())`
Well, better formatted of course as we might have a long list of prop names.

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@m-ruslan almost there

**Comment by m-ruslan** [comment]
@OleksiyRudenko, I'm not sure whether I've made it right this time)
Could you please review?

**Review by OleksiyRudenko — APPROVED** [review]
@m-ruslan now it is explicitly clear what building blocks `makeMessage` uses to construct an object presentation. Well done!

**Comment by m-ruslan** [comment]
@OleksiyRudenko thank you)

**Pull request merged**